WebApplication/ClientApp/src/actions/adoptWithParamsActions.js

```diff
@@ -0,0 +1,48 @@
+import { addError, addLog } from './notificationActions';
+import { Jobs } from '../JobManager';
+import { showAdoptWithParametersProgress, updateActiveTabIndex } from './uiFlagsActions';
+import { updateActiveProject } from '../actions/projectListActions';
+import { addOrUpdateProject } from './projectListActions';
+import { adaptParameters, updateParameters } from './parametersActions';
+
+export const adoptProjectWithParameters = (parameters) => async (dispatch) => {
+    dispatch(addLog('adoptProjectWithParameters invoked'));
+
+    const jobManager = Jobs();
+
+    // launch progress dialog immediately before we started connection to the server
+    dispatch(showAdoptWithParametersProgress(true));
+
+    try {
+        await jobManager.doAdoptWithParameters(parameters,
+            // start job
+            () => {
+                dispatch(addLog('JobManager: HubConnection started for adopt project with params'));
+            },
+            // onComplete
+            (tuple) => {
+                dispatch(addLog('JobManager: Adopt project with paramscReceived onComplete'));
+
+                const project = tuple.item1;
```

Comment:
we may consider do it more nice :) Using type instead of tuple

Reply:
Improved based on Miros changes, we may still conside rmaking the mutation on BE as it looks a bit odd on client side